src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/util/ObjectGraphUtils.java

```diff
@@ -106,6 +109,16 @@ private ObjectTracer(final SqueakImageContext image) {
             addObjectsFromTruffleFrames();
         }
 
+        private void cancelPendingTrace(final SqueakImageContext image, final ArrayDeque<AbstractSqueakObjectWithClassAndHash> marked) {
+            AbstractSqueakObjectWithClassAndHash currentObject;
+            // Unmark known marked objects.
+            while ((currentObject = marked.pollLast()) != null) {
```

Comment:
I'd just use a for-loop here and let the GC clean up the marked deque. No need to remove all elements one by one.